optuna_dashboard/ts/trialFilter.ts

```diff
@@ -192,3 +192,58 @@ export const useObjectiveAndUserAttrTargets = (
   )
   return [targetList, selectedTarget, setTargetIdent]
 }
+
+export const useObjectiveAndUserAttrTargetsFromStudies = (
+  studies: StudyDetail[]
+): [Target[], Target, (ident: string) => void] => {
+  const defaultTarget = new Target("objective", 0)
+  const [selected, setTargetIdent] = useState<string>(
+    defaultTarget.identifier()
+  )
+  const minDirections = useMemo<number>(() => {
+    return studies.reduce((acc, study) => {
+      return Math.min(acc, study.directions.length)
+    }, Number.MAX_VALUE)
+  }, [studies])
+
+  const intersect = (arrays: AttributeSpec[][]) => {
+    const deepEqual = (obj1: AttributeSpec, obj2: AttributeSpec) => {
+      return JSON.stringify(obj1) === JSON.stringify(obj2)
```

Comment:
<img width="515" alt="Screenshot 2023-07-28 18 00 25" src="https://github.com/optuna/optuna-dashboard/assets/5564044/27618b4e-a95b-42ec-831f-7f6f9f434dcb">

How about directly checking the `obj1.key === obj2.key`?

Reply:
Fixed it.